### PR TITLE
Remove state migration from westend runtime.

### PR DESCRIPTION
**Pull request by cheme**
State migration did pass on westend and can be remove from the runtime.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2405150

**Review by bkchr — APPROVED** [review]
The pallet stored data? We should write some cleanup function.

**Comment by cheme** [comment]
Oh, did not think about it :+1: . The data stored is small though (just the current position) so could make sense to keep it just to know it was migrated. But I think I will clean it up.

**Comment by cheme** [comment]
I did write a clean up function, but did not find a good way to also remove the dependency from the runtime.

**Comment by cheme** [comment]
I am hesitant to just hardcode the storage value of the storage items (and remove directly with the storage api) and remove the module.
But then it is really not auditable code.

File: runtime/westend/src/lib.rs [hunk 1]
```diff
@@ -1908,43 +1908,35 @@ mod remote_tests {
...
-				AutoLimits::<Runtime>::put(Some(MigrationLimits { item: 160, size: 204800 }));
-				log::info!("Automatic trie migration started.");
-				<Runtime as frame_system::Config>::DbWeight::get().reads_writes(2, 1)
-			} else {
-				log::info!("Automatic trie migration not started.");
-				<Runtime as frame_system::Config>::DbWeight::get().reads(2)
-			}
+			MigrationProcess::<Runtime>::kill();
+			AutoLimits::<Runtime>::kill();
+			SignedMigrationMaxLimits::<Runtime>::kill();
```
Comment (bkchr) [inline runtime/westend/src/lib.rs:1929]: You can use `storage_alias` to have the definition of them without including the pallet.

File: runtime/westend/src/lib.rs [hunk 2]
```diff
@@ -1908,43 +1892,64 @@ mod remote_tests {
...
+		fn name() -> &'static str {
+			"StateTrieMigration"
+		}
+		fn module_name() -> &'static str {
+			"pallet_state_trie_migration"
+		}
+		fn crate_version() -> frame_support::traits::CrateVersion {
+			frame_support::traits::CrateVersion { major: 4u16, minor: 0u8, patch: 0u8 }
+		}
+	}
```
Comment (bkchr) [inline runtime/westend/src/lib.rs:1918]:
```suggestion
```

File: runtime/westend/src/lib.rs [hunk 3]
```diff
@@ -1908,43 +1892,64 @@ mod remote_tests {
...
+
+	#[storage_alias]
+	type AutoLimits<T> = StorageValue<Pallet<T>, Option<MigrationLimits>, ValueQuery>;
```
Comment (bkchr) [inline runtime/westend/src/lib.rs:1921]:
```suggestion
	type AutoLimits = StorageValue<StateTrieMigration, Option<MigrationLimits>, ValueQuery>;
```

File: runtime/westend/src/lib.rs [hunk 4]
```diff
@@ -1908,43 +1892,64 @@ mod remote_tests {
...
+
+	// Actual type of value is `MigrationTask<T>`, putting a dummy
+	// one to avoid the trait constraint on T.
+	// Since we only use `kill` it is fine.
+	#[storage_alias]
+	type MigrationProcess<T> = StorageValue<Pallet<T>, u32, ValueQuery>;
+
+	#[storage_alias]
+	type SignedMigrationMaxLimits<T> = StorageValue<Pallet<T>, MigrationLimits, OptionQuery>;
```
Comment (bkchr) [inline runtime/westend/src/lib.rs:1930]:
```suggestion
	type MigrationProcess = StorageValue<StateTrieMigration, u32, ValueQuery>;

	#[storage_alias]
	type SignedMigrationMaxLimits = StorageValue<StateTrieMigration, MigrationLimits, OptionQuery>;
```

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2408632

File: runtime/westend/src/lib.rs [hunk 5]
```diff
@@ -1926,43 +1910,47 @@ mod remote_tests {
...
 	use sp_std::prelude::*;
 
+	#[storage_alias]
+	type AutoLimits = StorageValue<StateTrieMigration, Option<MigrationLimits>, ValueQuery>;
+
+	// Actual type of value is `MigrationTask<T>`, putting a dummy
+	// one to avoid the trait constraint on T.
+	// Since we only use `kill` it is fine.
+	#[storage_alias]
+	type MigrationProcess = StorageValue<StateTrieMigration, u32, ValueQuery>;
```
Comment (ggwpez) [inline runtime/westend/src/lib.rs:1928]: I think you can also just delete the pallet prefix instead of defining each alias but this also works obviously.

Comment (cheme) [reply runtime/westend/src/lib.rs:1928]: Thanks for the tip, did not think about this approach, with 3 value only I will still keep this version.

**Comment by cheme** [comment]
bot merge

**Pull request merged**